R/zchunk_LB162.ag_prodchange_R_C_Y_GLU_irr.R

```diff
@@ -41,67 +45,460 @@ module_aglu_LB162.ag_prodchange_R_C_Y_GLU_irr_DISABLED <- function(command, ...)
+    # Get the set of possible combinations and join in the YieldRates from L162.agBio_YieldRate_R_C_Ysy_GLU_irr.
+    # For combinations not covered by L162.agBio_YieldRate_R_C_Ysy_GLU_irr, fill in the values from the default table above.
+    L161.ag_irrProd_Mt_R_C_Y_GLU %>%
+      mutate(Irr_Rfd = "IRR") %>%
+      bind_rows(mutate(L161.ag_rfdProd_Mt_R_C_Y_GLU, Irr_Rfd = "RFD"))  %>%
+      select(GCAM_region_ID, GCAM_commodity, GLU, Irr_Rfd) %>%
+      dplyr::distinct() ->
+      L162.ag_Prod_Mt_R_C_Y_GLU_irr
+
+    L162.ag_Prod_Mt_R_C_Y_GLU_irr %>%
+      left_join(L162.agBio_YieldRate_R_C_Ysy_GLU_irr, by = c("GCAM_region_ID", "GCAM_commodity", "GLU", "Irr_Rfd")) %>%
+      # NA's include NA years, address
+      tidyr::complete(year = SPEC_AG_PROD_YEARS, nesting(GCAM_region_ID, GCAM_commodity, GLU, Irr_Rfd)) ->
+      # store in a table for further processing
+      L162.ag_YieldRate_R_C_Y_GLU_irr
+
+    # Subset to only the complete cases - group by region-commodity-glu-irrigation and keep only the members with non-na entries
+    # for every year
+    L162.ag_YieldRate_R_C_Y_GLU_irr %>%
+      # isolate the incomplete rows, wipe out their existing data, and pull in default yield rates
+      group_by(GCAM_region_ID, GCAM_commodity, GLU, Irr_Rfd) %>%
+      filter(!any(is.na(YieldRate))) %>%
+      ungroup() ->
+      L162.ag_YieldRate_R_C_Y_GLU_irr_completecases
+
+    # Isolate the incomplete cases, fill in default yield rates for each year, and join to the complete cases
+    L162.ag_YieldRate_R_C_Y_GLU_irr %>%
+      # isolate the incomplete rows, wipe out their existing data, and pull in default yield rates
+      group_by(GCAM_region_ID, GCAM_commodity, GLU, Irr_Rfd) %>%
+      filter(any(is.na(YieldRate))) %>%
+      ungroup() %>%
+      select(-YieldRate) %>%
+      left_join_error_no_match(L162.defaultYieldRate, by = c("year", "GCAM_commodity")) %>%
+      rename(YieldRate = defaultRate) %>%
+      # incorporate back into main data frame
+      bind_rows(L162.ag_YieldRate_R_C_Y_GLU_irr_completecases) ->
+      L162.ag_YieldRate_R_C_Y_GLU_irr
+
+
+    # Step 3: Expand to future years by applying the default rate in each year
+    L162.ag_YieldRate_R_C_Y_GLU_irr %>%
+      tidyr::complete(year = c(max(HISTORICAL_YEARS),FUTURE_YEARS), nesting(GCAM_region_ID, GCAM_commodity, GLU, Irr_Rfd)) %>%
+      left_join_error_no_match(L162.defaultYieldRate, by = c("year", "GCAM_commodity")) %>%
+      # replace NA's - which correspond to years we just filled in - with the default yield for that year we just joined
+      group_by(GCAM_region_ID, GCAM_commodity, GLU, Irr_Rfd, year) %>%
+      mutate(YieldRate = replace(YieldRate,
+                                 is.na(YieldRate),
+                                 defaultRate)) %>%
+      ungroup() %>%
+      select(-defaultRate) ->
+      L162.ag_YieldRate_R_C_Y_GLU_irr
+
+
+    # Step 4: Write out biomass yields
+    # Essentially repeating steps 1-3 with biomass only.
```

Comment:
Actually, I'm pretty sure the only reason this is done separately in the old DS is because some intermediate products don't have Biomass.

So I think I should be able to streamline this into a single pipeline with maybe one intermediate pipeline ahead of it, and then separate out the ag and bio for outputs at the end. Testing now.